spark/src/test/scala/com/gu/tableversions/spark/VersionContextSpec.scala

```diff
@@ -277,6 +277,16 @@ class VersionContextSpec extends FlatSpec with Matchers with SparkHiveSuite {
     tableUpdate.userId shouldBe userId
   }
 
+  "withVersionedScheme" should "preserve all URI parts" in {
+
+    val uri = new URI("s3", "bucketName", "/some/path/date=2019-01-01", "someQueryString", "someFragment")
+
+    val expectedUri =
+      new URI(VersionedFileSystem.scheme, "bucketName", "/some/path/date=2019-01-01", "someQueryString", "someFragment")
+
+    SparkSupport.setVersionedScheme(uri) shouldBe expectedUri
+  }
+
```

Comment:
It might be an overkill to have a test just for this, but the current tests didn't catch the problem because the local filesystem doesn't use all the parts of a uri

Reply:
It's definitely not overkill! :) In fact, if we add some common code for manipulating URIs, I'd like to test the heck out of it using property based tests... as you say, it's proved a source of errors before.